pageserver/src/tenant/storage_layer/layer_desc.rs

```diff
@@ -57,6 +57,34 @@ impl std::fmt::Display for PersistentLayerKey {
     }
 }
 
+impl From<ImageLayerName> for PersistentLayerKey {
+    fn from(image_layer_name: ImageLayerName) -> Self {
+        Self {
+            key_range: image_layer_name.key_range,
```

Comment:
Weird that we never needed this before -- are there some locations that do this by hand which could be updated to use the From?

Reply:
Can't find any.

We do have `PersistentLayerDesc::from_filename`, but, that's an enriching function because a PersistentLayerDESC carries more information than a filename, whereas a PersistentLayerKEY carries equivalent information to a file anme.